pkg/ko/build/gobuild.go

```diff
@@ -158,19 +233,32 @@ func (gb *gobuild) Build(s string) (v1.Image, error) {
 	}
 	defer os.RemoveAll(filepath.Dir(file))
 
-	// Construct a tarball with the binary.
-	layerBytes, err := tarBinary(file)
+	var layers []v1.Layer
+	// Create a layer from the kodata directory under this import path.
+	dataLayerBytes, err := tarKoData(s)
 	if err != nil {
 		return nil, err
 	}
+	dataLayer, err := tarball.LayerFromOpener(func() (io.ReadCloser, error) {
+		return v1util.NopReadCloser(bytes.NewBuffer(dataLayerBytes)), nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	layers = append(layers, dataLayer)
```

Comment:
I don't like that this always adds a layer, even if there's no kodata.

Reply:
tl;dr I'm not going to lose much sleep over it.

I debated adding it to the same tarball, but by splitting things out, you should be able to iterate on either static assets or your binary faster.  Eliding this seems unnecessary since when it's not needed it'll disappear in the latency of writing the binary layer (or probably even the config, it is so small).